dashboard/utils.py

```diff
@@ -219,7 +229,7 @@ def has_paid_final_grade(self, edx_course_key):
         Returns:
             bool: whether a frozen final grade exists
         """
-        return self.final_grade_qset.paid_on_edx().for_course_run_key(edx_course_key).exists()
+        return self.has_final_grade(edx_course_key) and self.has_paid(edx_course_key)
```

Comment:
@gsidebo what you say is right, but it is more complicated than that:
a FA user must considered as "not paid" even if the "paid on edX" is True.